lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -91,6 +94,45 @@ sub view ($self) {
     $self->viewimg;
 }
 
+sub _checkout_needledir_or_default {
```

Comment:
This whole function should likely return early unless this feature is turned on in the settings. I imagine that the performance impact and additional storage usage are not generally neglectable and our openQA instance is at its limits as-is. Besides, the code might not work depending on the state of the Git checkout which might cause problems in production use (e.g. the Git checkout might currently be locked because some other operation is performed or it is in an intermediate state because currently a user saves a needle via the needle editor or simply the hard-coded remote names don't match).